### PR TITLE
Clickable country name in profile

### LINKED ISSUE #23324: User profile flag label is not clickable unlike web
### Type

Other

### Bug description

Just nitpicking the differences between lazer and web behavior.

<sub>Mostly fulfilling my "Make osu! repo more friendly for new contributors" goal.</sub>

### Screenshots or videos

| Lazer | Web |
| --- | --- |
| osu!_YgKZ07CKI5 | firefox_lHcm0y79GM |

### Version

Since 2023.207.0 / https://github.com/ppy/osu/commit/ef7812412b5eadf301de7aa2d7e090f450d76d65

### Logs

N/A

### PR DESCRIPTION
**Pull request by rltn**
This change resolves #23324 by making the country name in profile clickable.

### CODE REVIEW

File: osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs [hunk 1]
```diff
@@ -229,5 +232,33 @@ public ProfileCoverBackground()
...
+                    rankingsOverlay?.ShowCountry(countryCode);
+                };
+
+                Child = new OsuSpriteText
+                {
+                    Font = Font,
+                    Text = countryCode.GetDescription(),
+                };
+            }
+        }
```
Comment (ItsShamed) [inline osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs:262]: You don't need to create a new class for this. You can just create a new instance of `OsuHoverContainer` and use the object initializer syntax to initialize `Action`, `AutoSizeAxes` and `Child`

**Review by ItsShamed — CHANGES_REQUESTED** [review]
Also, please make sure to run the `InspectCode.ps1` script (or `InspectCode.sh` if you are on a *nix-type system) and follow the linting of your IDE to be on par with the code quality standards.

File: osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs (hunk 1 above)
Comment (rltn) [reply osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs]: Thanks, i understand the project stucture much better right now. But i have a question about the code inspection. The JetBrains Inspect Code creates a huge report file and i'm pretty sure it shouldn't do that.
image

Comment (ItsShamed) [reply osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs]: This is to be studied with ReSharper, however it is paid AFAIK. But you can take a glance at the file to see what the issues are (shouldn't be difficult to figure out what all of this means), or you can search online for softwares that can parse this.

File: osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs [hunk 2]
```diff
@@ -156,13 +162,20 @@ private void load(OverlayColourProvider colourProvider, OsuConfigManager configM
...
-                                                        userCountryText = new OsuSpriteText
+                                                        userCountryContainer = new OsuHoverContainer
                                                         {
-                                                            Font = OsuFont.GetFont(size: 14f, weight: FontWeight.Regular),
                                                             Margin = new MarginPadding { Left = 5 },
                                                             Origin = Anchor.CentreLeft,
                                                             Anchor = Anchor.CentreLeft,
-                                                        }
+                                                            AutoSizeAxes = Axes.Both,
+                                                            Children = new Drawable[]
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Header/TopHeaderContainer.cs:171]: Can use `Child` not `Children`.

**Review by peppy — APPROVED** [review]
Work well enough

**Pull request merged**